netket/graph/edgeless.py

```diff
@@ -0,0 +1,28 @@
+from graph import Graph
+
+import numpy as _np
+import networkx as _nx
+
+
+class Edgeless(Graph):
+    r"""A set graph (collection of unconnected vertices)."""
+
+    def __init__(self, n_nodes):
+        """
+    	Constructs a new set of given number of vertices.
+
+		Args:
+		    n_nodes: The number of nodes.
+
+		Examples:
+		    A 10-site set:
+
+		    >>> import netket
+		    >>> g=netket.graph.Edgeless(10)
+		    >>> print(g.n_nodes)
+		    10
+    	"""
+        super().__init__([(i, i) for i in range(n_nodes)])
+        edgelessgraph = _nx.MultiGraph()
+        edgelessgraph.add_nodes_from(self.graph)
+        self.graph = edgelessgraph
```

Comment:
i don't see why this `__init__` function calls the base class constructor with one self-loop for each node in the first line and then immediately discards the `networkx` graph that `Graph.__init__` created from that edge list and assigns a different one.

That seems like it could be simplified, for example by allowing the construction of `Graph` directly from a `networkx` graph. That is in the end what the `Graph` class is now: A wrapper around a `networkx` graph (while `AbstractGraph` is the more general interface). So making it possible to directly create it that way seems logical to me.

Reply:
@femtobit yes, I was thinking the same to solve the automorphisms ordering problem. However, the translation from a `Graph` to a `MultiGraph` is always necessary (to be able to represent colours). As it stands right now, the `Graph` class could have `nodes` and `edges` as arguments so that the `Edgeless` graph can be constructed by passing only `nodes`. Also, users could create graphs from `edges` only, and `hypergrid` constructors can create graphs from either `edges` or `nodes` or both. What do you think?

Reply:
yes it seems to me that we should just have a way to convert a generic networkx graph into a ```netket.Graph``` object, then all the other cases are special instantations of this wrapper

Reply:
@gcarleo Ok, so, do I write `Graph` to accept a `networkx` graph and convert it to a `MultiGraph` object to handle colours? If this is true, do I create a `CustomGraph` class that can receive `nodes` and/or `edges` for people that want to enter them explicitly?

Reply:
@VolodyaCO : The way I see it, you should implement a single class (call it ```netket.graph.NetworkX``` or something, that takes a networkx object and wraps it into something that satisfies the ```AbstractGraph``` requirements and methods. 

Then, ```Graph``` is another class that should work accepting nodes and or edges, and basically have the same interface that it has now ```CustomGraph```. Notice however that internally you can just implement ```Graph``` as something that does two thinks:

1. Takes the edges and or nodes and creates a networkx object (call it ```n_graph```)
2. Returns ```netket.graph.NetWorkx(n_graph)```, so that all methods are automatically implemented...

Reply:
In this sense, ```netket.graph.Graph``` is not even a class but it's just a pure function

Reply:
Good, I'll do that today.

Reply:
I implemented the `Graph` as a function, `Edgeless` as a function, `Hypergrid` as a class. @gcarleo , do you want me to treat `Hypercube`, `Square` and `Chain` as functions as well? Implementing them as a class is also not needed in my opinion...

Reply:
yes it makes sens